### ISSUE TITLE
Inconsistent indentation of long lines

### ISSUE THREAD

**Issue by micheleCTDE** [uncrustify/uncrustify#3105]
uncrustify b51e2c22 + additional code for quick dumping (see this [branch](https://github.com/micheleCTDE/uncrustify/tree/dump-cycle) - rebase on top of master if necessary).

Problem: indentation of the same long line is different when a few extra characters are added.

The configuration uses
```
indent_continue = -6
```
to indent the continuation of a long line.
In the attached file, the output in test1 folder is correctly intended, while the output in test2 folder is wrongly indented. The difference in the source files is just 3 characters.
The output in test 1 (correct) is:
```
  connectFor(&m_timer, averylongfieldvariable, TQT_SIGNAL(timeoutVeryVeryLong()),
        TQT_SIGNAL(timeout()), this, TQT_SLOT(slotTimeout()));
```
and in test2 (wrong) is:
```
  connect(&m_timer, averylongfieldvariable, TQT_SIGNAL(timeoutVeryVeryLong()), TQT_SIGNAL(
          timeout()), this, TQT_SLOT(slotTimeout()));
```
Notice the different indentation of the continuation line.

To reproduce run: 
```
uncrustify -c test.cfg -f t1.cpp -o o1.cpp --dump -L 21,22,31,71,88,102 2>log1.log
uncrustify -c test.cfg -f t2.cpp -o o2.cpp --dump -L 21,22,31,71,88,102 2>log2.log
```

I will problably work on this myself in the near future. Will prepare a PR when ready.

[inconsistent_continuation.zip](https://github.com/uncrustify/uncrustify/files/6054655/inconsistent_continuation.zip)

**Comment by micheleCTDE** [comment]
Additional info: the problem is caused during the second iteration of the final do-while loop. 
The first iteration splits the code correctly (see dump_4.log), then when calling indent_text() on the second iteration, the intentation is messed up.

**Comment by micheleCTDE** [comment]
Hi Guy, the --dump option is a dirty-n-quick option I added to my local fork to get extra info for troubleshooting. It creates all those dump_X.log files you find in the attachment. You can find the code for such option [here](https://github.com/micheleCTDE/uncrustify/tree/dump-cycle). Since it is quick-n-dirty I didn't craete a PR, it would need some polishing and rework. If you think this could be useful, I can prepare something after April.

**Comment by micheleCTDE** [comment]
For the record, I am working on a fix for this

**Comment by micheleCTDE** [comment]
PR #3982 is a fix for this issue. The problem was that a split on an open brace was increasing the indentation level, even if indent_continue was set to negative values which are supposed to be absolute

**Comment by micheleCTDE** [comment]
Solved by PR #3982
